dotCMS/src/main/java/com/dotmarketing/webdav/DotWebdavHelper.java

```diff
@@ -920,20 +920,20 @@ public void setResourceContent(String resourceUri, InputStream content,	String c
                 FileUtil.fastCopyUsingNio(inputChannel, outputChannel);
                 Logger.debug(this, "WEBDAV fileName:" + fileName + " : File size:" + fileData.length() + " : " + fileData.getAbsolutePath());
                 
-                //This is only needed for Cyberduck, when you right click>New File
-				if(fileData.length() == 0 && !Config.getBooleanProperty("CONTENT_ALLOW_ZERO_LENGTH_FILES", false) && HttpManager.request().getUserAgentHeader().contains("Cyberduck")){
+                //Avoid uploading an empty file
+				if(fileData.length() == 0 && !Config.getBooleanProperty("CONTENT_ALLOW_ZERO_LENGTH_FILES", false) ){
 					Logger.warn(this, "The file " + folder.getPath() + fileName + " that is trying to be uploaded is empty. A byte will be written to the file because empty files are not allowed in the system");
-					FileUtil.write(fileData, " ");
+					FileUtil.write(fileData, "~DOTEMPTY");
 				}
```

Comment:
this IF block can be extracted to a method and make the call here and in the next occurrence around line 974